src/analysis/processing/qgsalgorithmcellstatistics.cpp

```diff
@@ -327,22 +362,436 @@ QVariantMap QgsCellStatisticsAlgorithm::processAlgorithm( const QVariantMap &par
+  }
+  mOutputRasterDataProvider->setEditable( false );
+}
+
+//
+//QgsCellStatisticsPercentRankFromValueAlgorithm
+//
+QString QgsCellStatisticsPercentRankFromValueAlgorithm::displayName() const
+{
+  return QObject::tr( "Cell stack percentrank from value" );
+}
+
+QString QgsCellStatisticsPercentRankFromValueAlgorithm::name() const
+{
+  return QObject::tr( "cellstackpercentrankfromvalue" );
+}
+
+QStringList QgsCellStatisticsPercentRankFromValueAlgorithm::tags() const
+{
+  return QObject::tr( "cell,pixel,statistic,percentrank,rank,percent,value" ).split( ',' );
+}
+
+QString QgsCellStatisticsPercentRankFromValueAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "The Cell stack percentrank from value algorithm returns the cell-wise percentrank value of a stack of rasters based on a single input value."
+                      "and writes them to an output raster. "
+                      "At each cell location, the specified value is used ranked among the respective values in the stack of all overlaid and sorted cell values of the input rasters. "
+                      "For values outside of the the stack value distribution, the algorithm returns NoData because the value cannot be ranked among the cell values.\n\n"
+                      "There are two methods for percentile calculation:"
+                      "<ul> "
+                      "   <li>Inclusive linearly interpolated percent rank (PERCENTRANK.INC)</li>"
+                      "   <li>Exclusive linearly interpolated percent rank (PERCENTRANK.EXC)</li>"
+                      "</ul> "
+                      "The linear interpolation method return the unique percent rank for different values. Both interpolation "
+                      "methods are following the methods implemented by LibreOffice or Excel. \n\n"
+                      "The output raster's extent and resolution is defined by a reference "
+                      "raster. If the input raster layers that do not match the cell size of the reference raster layer will be "
+                      "resampled using nearest neighbor resampling.  NoData values in any of the input layers will result in a NoData cell output if the Ignore NoData parameter is not set. "
+                      "The output raster data type will always be Float32." );
+}
+
+QgsCellStatisticsPercentRankFromValueAlgorithm *QgsCellStatisticsPercentRankFromValueAlgorithm::createInstance() const
+{
+  return new QgsCellStatisticsPercentRankFromValueAlgorithm();
+}
+
+void QgsCellStatisticsPercentRankFromValueAlgorithm::addSpecificAlgorithmParams()
+{
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "METHOD" ), QObject::tr( "Method" ), QStringList() << "Inclusive linear interpolation (PERCENTRANK.INC)" << "Exclusive linear interpolation (PERCENTRANK.EXC)", false, 0, false ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "VALUE" ), QObject::tr( "Value" ), QgsProcessingParameterNumber::Double, 10, false ) );
```

Comment:
again, is 10 the correct default here?

Reply:
Yes this is correct. In fact the value would highly depend on the value ranges of the entered rasters. I guess 10 is a reasonable example value.